### PR TITLE
Fix source position for lambda parameters

### PR DESCRIPTION
**Pull request by mickaelistria**
Fix https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2236

## Author checklist

- [ ] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by mickaelistria** [comment]
Added a test.

**Comment by mickaelistria** [comment]
@jjohnstn @rgrunber @akurtakov Can one of you please review/merge?

**Comment by mickaelistria** [comment]
> As lambdas are Java 8 features, shouldn't the test go into ASTConverter18Test ?

OK, I'll change that.

**Comment by mickaelistria** [comment]
Done.

File: org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/VariableBinding.java [hunk 1]
```diff
@@ -240,17 +240,16 @@ private JavaElement getUnresolvedJavaElement() {
 		int sourceStart;
 		int sourceLength;
 		int modifiers = 0;
-		if (localVar instanceof SingleVariableDeclaration) {
-			sourceStart = localVar.getStartPosition();
-			sourceLength = localVar.getLength();
-			final SingleVariableDeclaration singleVariableDeclaration = (SingleVariableDeclaration) localVar;
+		sourceStart = localVar.getStartPosition();
```
Comment (akurtakov) [inline org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/VariableBinding.java:243]: Please combine the initialization of sourceStart/sourceLength with the declarations a line above for the sake of better readability.

Comment (mickaelistria) [reply org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/VariableBinding.java]: OK, done.

**Pull request merged**